model/configuration.py

```diff
@@ -769,3 +774,567 @@ def excluded_audio_data_sources(cls, _db):
         if value is None:
             value = cls.EXCLUDED_AUDIO_DATA_SOURCES_DEFAULT
         return value
+
+
+class ExternalIntegrationOwner(object):
+    """Interface allowing to get access to an external integration"""
+
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def external_integration(self, db):
+        """Returns an external integration owned by this object
+
+        :param db: Database session
+        :type db: sqlalchemy.orm.session.Session
+
+        :return: External integration owned by this object
+        :rtype: core.model.configuration.ExternalIntegration
+        """
+        raise NotImplementedError()
+
+
+class CollectionExternalIntegrationOwner(object):
```

Comment:
Since this is like `HasExternalIntegration` for things that can have a different `ExternalIntegration` for every `Collection`, I suggest calling it `HasExternalIntegrationPerCollection`.

Reply:
Changed it to `HasExternalIntegrationPerCollection`